File: frontend/src/schema/tweet.ts
import { z } from 'zod';

export const TweetSchema = z.object({
  id: z.string(),
  content: z.string(),
  likeCount: z.number(),
  createdAt: z.date(),
  doubtRating: z.number(),
  media: z.array(z.string()),
  quotedTweetId: z.string().optional(),
  userId: z.string(),
  aiToolsMentioned: z.array(z.string())
});

export type Tweet = z.infer<typeof TweetSchema>;

File: backend/app/schema/tweet.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class Tweet(BaseModel):
    id: str
    content: str
    like_count: int
    created_at: datetime
    doubt_rating: float
    media: List[str]
    quoted_tweet_id: Optional[str]
    user_id: str
    ai_tools_mentioned: List[str]